src/main/scala/com/yotpo/metorikku/output/writers/jdbc/JDBCQueryWriter.scala

```diff
@@ -0,0 +1,46 @@
+package com.yotpo.metorikku.output.writers.jdbc
+
+import java.sql.{Date, DriverManager}
+
+import com.yotpo.metorikku.configuration.outputs.JDBC
+import com.yotpo.metorikku.output.MetricOutputWriter
+import org.apache.spark.sql.DataFrame
+
+import scala.collection.mutable
+
+class JDBCQueryWriter(props: Map[String, String], config: Option[JDBC]) extends MetricOutputWriter {
+
+  case class JDBCQueryProperties(query: String)
+
+  val options = JDBCQueryProperties(props("query"))
+
+  override def write(dataFrame: DataFrame): Unit = {
+    config match {
+      case Some(config) =>
+        dataFrame.foreachPartition(partition => {
+          val conn = DriverManager.getConnection(config.connectionUrl, config.user, config.password)
+          val stmt = conn.prepareStatement(options.query)
+
+          partition.foreach(row => {
+            for (i <- 1 to row.size) {
+              val v = row.get(i-1)
+              v match {
+                case v: String => stmt.setString(i, v.asInstanceOf[String])
+                case v: Date => stmt.setDate(i, v.asInstanceOf[Date])
+                case v: Boolean => stmt.setBoolean(i, v.asInstanceOf[Boolean])
+                case v: Int => stmt.setInt(i, v.asInstanceOf[Int])
+                case v: Double => stmt.setDouble(i, v.asInstanceOf[Double])
+                // Handling all random object types and NULLs
+                case _ => stmt.setObject(i, v)
```

Comment:
Don't you want to save it as JSON perhaps?

Reply:
The "_" ? definitely not JSON, maybe string but both solutions are guessing. I prefer to leave it to the default behaviour and overwrite it only if needed.